### PR TITLE
[BugFix] Fix IllegalState Exception thrown from TabletScheduler

### PR DESCRIPTION
**Pull request by gengjun-git**
Fixes
```
java.lang.IllegalStateException: 6261562
        at com.google.common.base.Preconditions.checkState(Preconditions.java:510) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.catalog.TabletInvertedIndex.getReplica(TabletInvertedIndex.java:567) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.DiskAndTabletLoadReBalancer.tryToBalanceTablet(DiskAndTabletLoadReBalancer.java:1244) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.DiskAndTabletLoadReBalancer.balanceTablet(DiskAndTabletLoadReBalancer.java:1179) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.DiskAndTabletLoadReBalancer.balanceBackendTablet(DiskAndTabletLoadReBalancer.java:1065) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.DiskAndTabletLoadReBalancer.selectAlternativeTabletsForCluster(DiskAndTabletLoadReBalancer.java:105) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.Rebalancer.selectAlternativeTablets(Rebalancer.java:76) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.TabletScheduler.selectTabletsForBalance(TabletScheduler.java:1281) ~[starrocks-fe.jar:?]
        at com.starrocks.clone.TabletScheduler.runAfterCatalogReady(TabletScheduler.java:377) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:73) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) ~[starrocks-fe.jar:?]
```
Remove the assert in `getReplica`, check nullable in the caller.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|8|87.50%|[1902]|

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -569,7 +569,6 @@ public void deleteReplica(long tabletId, long backendId) {
     public Replica getReplica(long tabletId, long backendId) {
         readLock();
         try {
-            Preconditions.checkState(tabletMetaMap.containsKey(tabletId), tabletId);
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:572]: why remove this?

File: fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java [hunk 2]
```diff
@@ -927,7 +927,7 @@ private Map<Pair<Long, Long>, Set<Long>> getPartitionTablets(long beId, TStorage
 
             if (pathHash != -1) {
                 Replica replica = GlobalStateMgr.getCurrentInvertedIndex().getReplica(tabletId, beId);
-                if (replica.getPathHash() != pathHash) {
+                if (replica == null || replica.getPathHash() != pathHash) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java:930]: can the replica be null, we don't have the db lock hold here?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java:930]: No db lock hold here, the metadata is retrieved only from TabletInvertedIndex.

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:572]: remove this assert and leave it to the caller to check nullable.

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**